geps/gep-1494/index.md

```diff
@@ -0,0 +1,151 @@
+# GEP-1494: HTTP Auth in Gateway API
+
+* Issue: [#1494](https://github.com/kubernetes-sigs/gateway-api/issues/1494)
+* Status: Provisional
+
+(See status definitions [here](/geps/overview/#gep-states).)
+
+
+## TLDR
+
+Provide a method for configuring **Gateway API implementations** to add HTTP Authentication for north-south traffic. The method may also include Authorization config if practical. At the time of writing, this authentication is only for ingress traffic to the cluster.
+
+
+## Goals
+
+(Using the [Gateway API Personas](https://gateway-api.sigs.k8s.io/concepts/roles-and-personas/))
+
+* A way to configure a Gateway Implementation to perform Authentication (at least), with optional Authorization on behalf of Ana the Application Developer.
+
+* A way for Chihiro the Cluster Admin to configure a default Authentication and/or Authorization config for some set of HTTP or GRPC matching criteria.
+
+* Optionally, a way for Ana to have the ability to disable Authentication and/or Authorization for specific routes when needed, allowing certain routes to not be protected. This would probably need to work something like a default enabling at Gateway level, that can be specifically set at lower levels, but further design is TBD.
```

Comment:
I think it might be helpful to draw out some motivation which explains this particular goal. Then, if we're going to provide a mechanism for the developer to override something [that is security related] set by the cluster administrator, we may want to consider flags to enable this override that Chihiro has to explicitly turn on.

Reply:
This feels less like flags and more like the defaults/overrides of the policy attachment model.

Some example HTTP endpoints that may be opted out of a default AuthN policy could be `/login` (for flows other than client-cert mTLS, unauthenticated human users will often need to get redirected _somewhere_ to authenticate and acquire a credential), or `/health` (some service health-checking mechanisms may prefer to avoid the overhead of authentication for performance reasons).

Reply:
Interesting :thinking:  An explanation of the motivation seems helpful in any case and maybe some elaboration here on the goal since apparently we two had somewhat different interpretations of it.

Reply:
I've updated with some other wording, PTAL. In the interest of time, we could also move this to Deferred Goals, and get the basic setup handled first (with an eye to being able to handle something like this in the future).